sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/compatibility/SpringCloudAzureSpringBootVersionVerifier.java

```diff
@@ -66,52 +70,68 @@ private boolean springBootVersionMatches() {
         return false;
     }
 
-    private Boolean bootVersionFromManifest(String s) {
+    private Optional<Boolean> bootVersionFromManifest(String s) {
         String version = this.getVersionFromManifest();
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug("Version found in Boot manifest [" + version + "]");
         }
         if (!StringUtils.hasText(version)) {
             LOGGER.info("Cannot check Boot version from manifest");
-            return null;
+            return Optional.empty();
         } else {
-            return version.startsWith(stripWildCardFromVersion(s));
+            return Optional.of(version.startsWith(stripWildCardFromVersion(s)));
         }
     }
 
-    static String stripWildCardFromVersion(String version) {
+    private static String stripWildCardFromVersion(String version) {
         return version.endsWith(".x") ? version.substring(0, version.indexOf(".x")) : version;
     }
 
-    CompatibilityPredicate is2_6() {
-        return new CompatibilityPredicate() {
-            public String toString() {
-                return "Predicate for Boot 2.6";
-            }
-            public boolean isCompatible() {
-                try {
-                    Class.forName("org.springframework.boot.autoconfigure.data.redis.ClientResourcesBuilderCustomizer");
-                    return true;
-                } catch (ClassNotFoundException ex) {
-                    return false;
-                }
-            }
-        };
+    private CompatibilityPredicate is2_6() {
+        return new CompatibilityPredicate2_6();
+    }
+
+    private CompatibilityPredicate is2_5() {
+        return new CompatibilityPredicate2_5();
     }
 
-    CompatibilityPredicate is2_5() {
-        return new CompatibilityPredicate() {
-            public String toString() {
-                return "Predicate for Boot 2.5";
+    @FunctionalInterface
+    interface CompatibilityPredicate {
+
+        /**
+         * Compatible of the current spring-boot version
+         * @return the version supported or not
+         */
+        boolean isCompatible();
+    }
+
+    private static class CompatibilityPredicate2_5 implements CompatibilityPredicate {
```

Comment:
Why are we changing this to class? There is no need to materialize these anonymous class.